### ISSUE TITLE
Make checkpoint a struct

### ISSUE THREAD

**Issue by Charleslee522** [bosnet/sebak#98]
Now, checkpoint type is `string` so when we want to use source or target, we have to call `ParseCheckpoint` method.
And checkpoint is used frequently, performance is important.
I think if we define the `Checkpoint struct` and define related methods then it would be helpful for both performance and code quality.

```
type Checkpoint struct {
  Source string
  Target string
}
```

**Comment by leejames00** [comment]
If `Checkpoint` is struct, `Source` and `Target` types should be hash, not string.. am i right?

**Comment by Charleslee522** [comment]
@leejames00 It would be better :)

**Comment by anarcher** [comment]
Maybe it can be another approach... How about it?  But I am not sure that it makes to improve performance. `~_~` 

```
type Checkpoint struct {
    Subtracted string // or Hash
    Added string // or Hash 
}
```

**Comment by outsideris** [comment]
What is another approach? I didn't get it.

**Comment by astonshnam** [comment]
It would be better to name them `Source` and` Target`. Later, when we develop TC, we can implement other functions besides boscoin transactions.

**Comment by anarcher** [comment]
I confused that ‘Source’ as SourceCheckpoint :-) 

My approach is same as yours :-) haha

**Comment by anarcher** [comment]
```
type Checkpoint struct {
    Hash string // Composed value: "<Subtracted>-<Added>"
    Subtracted string
    Added string
}
```

**Comment by spikeekips** [comment]
Resolved by #140
